### ISSUE TITLE
NullReferenceException when trying to run AutoML regression experiment from code

### ISSUE THREAD

**Issue by superichmann** [dotnet/machinelearning#6558]
**System Information (please complete the following information):**
 - OS & Version: Microsoft Windows 10 Pro 10.0.19045 N/A Build 19045
 - ML.NET Version: 2.0.1
 - microsoft.ml.automl: 0.20.1
 - .NET Version: .NET 6.0

**Describe the bug**
When trying to run AutoML regression experiment on a table with data of 30 rows in QuestDB (implements Postgresql) the following exception occurs:

		Message	"`Object reference not set to an instance of an object.`"	string
		Source	"`Microsoft.ML.AutoML`"	string
		StackTrace	" `  at Microsoft.ML.AutoML.AutoMLExperiment.<RunAsync>d__26.MoveNext()`"	string
		TargetSite	`{Void MoveNext()}	System.Reflection.MethodBase {System.Reflection.RuntimeMethodInfo}`
		DeclaringType	{Name = "`<RunAsync>d__26" FullName = "Microsoft.ML.AutoML.AutoMLExperiment+<RunAsync>d__26`"}	System.Type {System.RuntimeType}

**To Reproduce**
Download [QuestDB](https://questdb.io/)
Create table `"CREATE TABLE 'tests' (Ticker SYMBOL, H FLOAT, P_T FLOAT,timestamp TIMESTAMP) timestamp (timestamp) PARTITION BY MONTH;"`
Insert 30 rows to the table
IN C#: get the table data into IDataView with `mlContext.Data.CreateDatabaseLoader` and `using Npgsql;`
run the following code:
`ExperimentResult<RegressionMetrics> experimentResult = mlContext.Auto().CreateRegressionExperiment(20).Execute(dataview, "P_T");`
**Expected behavior**
The code should not throw an Exception and finish the experiment. **This code works perfectly on a large table with 28000 rows and 30 columns**

**Additional context**
I Tried running the experiment from the modelbuilder on csv and it works.

**Comment by superichmann** [comment]
I have tested same data csv load on C# code and got the same exception :(

**Comment by superichmann** [comment]
[questdb-query-1675793575444.csv](https://github.com/dotnet/machinelearning/files/10708876/questdb-query-1675793575444.csv)

**Comment by superichmann** [comment]
mmm if you want I cen test it just tell me how

**Comment by LittleLittleCloud** [comment]
Sure. The package is available on nightly build. You can find the nightly build feed on readme and simply install the latest version

**Comment by superichmann** [comment]
I tried running the instructions [here ](https://github.com/dotnet/machinelearning/blob/main/docs/building/windows-instructions.md) for half a day now and I got much errors in the type of `The source code won't be available via Source Link.` :((((((((

**Comment by LittleLittleCloud** [comment]
Looks like you are building from source? You don't have to do it if you just want to use the nightly build package.

Here's the instruction of how to add nightly build feed to your project
https://dev.azure.com/dnceng/public/_artifacts/feed/dotnet-libraries/connect/dotnet

**Comment by superichmann** [comment]
I have done [these ](https://dev.azure.com/dnceng/public/_artifacts/feed/dotnet-libraries/connect/visual%20studio) instructions and installed from NuGet manager from this source, still getting these versions :(
```
microsoft.ml\3.0.0-preview.23109.1
microsoft.ml.automl\0.21.0-preview.23109.1
```

**Comment by LittleLittleCloud** [comment]
@superichmann

Looks like you are on the right nightly build package. Does your code work now.

**Comment by superichmann** [comment]
No, still same exception.

As well I have tested on these versions since they are in the prealpha tick box on the regular NuGet feed... but I thought these are not the ones you meant :octocat:

**Comment by superichmann** [comment]
```
System.AggregateException
  HResult=0x80131500
  Message=One or more errors occurred. (Object reference not set to an instance of an object.)
  Source=System.Private.CoreLib
  StackTrace:
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at Microsoft.ML.AutoML.AutoMLExperiment.Run()
   at Microsoft.ML.AutoML.RegressionExperiment.Execute(IDataView trainData, ColumnInformation columnInformation, IEstimator`1 preFeaturizer, IProgress`1 progressHandler)
   at AutoML_Example.Program.Main(String[] args) in source\repos\ModelBuilder001\Program.cs:line 29

  This exception was originally thrown at this call stack:
    [External Code]

Inner Exception 1:
NullReferenceException: Object reference not set to an instance of an object.

```

**Comment by LittleLittleCloud** [comment]
Hmmm interesting, I test your dataset on main branch and that works

Here's my code

```csharp
public void AutoFit_QuestDB()
        {
            var context = new MLContext(1);
            context.Log += (o, e) =>
            {
                if (e.Source.StartsWith("AutoMLExperiment"))
                {
                    this.Output.WriteLine(e.RawMessage);
                }
            };
            var data = context.Data.LoadFromTextFile<QuestDB>("C:\\Users\\xiaoyuz\\Desktop\\questdb-query-1675793575444.csv", hasHeader: true, separatorChar: ',');
            var label = "P_T";
            var settings = new RegressionExperimentSettings
            {
                MaxModels = 100,
            };
            settings.Trainers.Remove(RegressionTrainer.StochasticDualCoordinateAscent);
            settings.Trainers.Remove(RegressionTrainer.LbfgsPoissonRegression);

            var result = context.Auto()
                .CreateRegressionExperiment(settings)
                .Execute(data, 10, label);

            Assert.True(result.BestRun.Results.Select(x => x.ValidationMetrics.RSquared).Min() > 0.70);
        }

class QuestDB
        {
            [LoadColumn(0)]
            [ColumnName("Ticker")]
            public string Ticker { get; set; }

            [ColumnName("H")]
            [LoadColumn(1)]
            public float H { get; set; }

            [ColumnName("P_T")]
            [LoadColumn(2)]
            public float PT { get; set; }

            [ColumnName("timestamp")]
            [LoadColumn(3)]
            public DateTime TimeStamp { get; set; }
        }
```

**Comment by LittleLittleCloud** [comment]
Can you give me a quick look of your csproj file

**Comment by superichmann** [comment]
omg funny! here it is:
```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net6.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.ML" Version="3.0.0-preview.23109.1" />
    <PackageReference Include="Microsoft.ML.AutoML" Version="0.21.0-preview.23109.1" />
  </ItemGroup>
</Project>
```

**Comment by LittleLittleCloud** [comment]
The csproj looks fine to me... can you also share the code of how you call into automl so I can try reproducing it, thanks

**Comment by superichmann** [comment]
Thanks for the effort man! I have changed your code to reproduce the problem (no folds, train test split or just to use all of the original data)

```
var context = new MLContext(1);
context.Log += (o, e) =>
{
    if (e.Source.StartsWith("AutoMLExperiment"))
    {
        Console.WriteLine(e.RawMessage);
    }
};
var data = context.Data.LoadFromTextFile<QuestDB>("training-data.csv", hasHeader: true, separatorChar: ',');
var splits = context.Data.TrainTestSplit(data, 0.2);
var train = splits.TrainSet;
var test = splits.TestSet;
var label = "P_T";
var settings = new RegressionExperimentSettings
{
    MaxExperimentTimeInSeconds = 5
};


var result = context.Auto()
    .CreateRegressionExperiment(settings)
    .Execute(train, label);
```

**Comment by LittleLittleCloud** [comment]
@superichmann 
Thanks, I'm able to reproduce the error and locate bug now. We'll push a fix really quickly.

As a workaround, you can either use cross-validation or explicilty set validation dataset

```
var trainTestSplit = context.Data.TrainTestSplit(train , 0.2);

...
var result = context.Auto()
    .CreateRegressionExperiment(settings)
    .Execute(trainTestSplit.Trainset, trainTestSplit.Testset, label);
```

Sorry for the trouble and let me know if that workaround works for you

**Comment by superichmann** [comment]
Thanks @LittleLittleCloud that works! You are awesome :)

**Comment by LittleLittleCloud** [comment]
fixed in #6560
